### PR TITLE
refactor: upgrade for cake4

### PR DESCRIPTION
**Pull request by josegonzalez**

### CODE REVIEW

**Comment by josegonzalez** [comment]
This one _almost_ works except there is a deprecated code usage in a dependency where the strategy is being tested (Tiny). Not sure what to do there.

Also this stupid phpstan error around treating a Behavior instance as the thing it is...

**Comment by josegonzalez** [comment]
Okay I hacked around it for now via a phpunit xml attribute, which is good enough until it gets fixed upstream (pull request [here](https://github.com/zackkitzmiller/tiny-php/pull/31)).

**Comment by ADmad** [comment]
Doesn't seem like `zackkitzmiller/tiny-php` is maintained anymore. We can just drop support for it.

File: .editorconfig [hunk 1]
```diff
@@ -1,18 +1,16 @@
 ; This file is for unifying the coding style for different editors and IDEs.
 ; More information at http://editorconfig.org
 
-root = true
+root = false
```
Comment (dereuromark) [inline .editorconfig]: Shouldnt this stay true? as this is the root config for this repo.

Comment (josegonzalez) [reply .editorconfig]: Pulled this from elsewhere...

File: composer.json [hunk 2]
```diff
@@ -31,11 +31,14 @@
         "source": "https://github.com/usemuffin/obfuscate"
     },
     "require": {
-        "cakephp/orm": "^3.5"
+        "cakephp/orm": "~4.0"
```
Comment (ADmad) [inline composer.json]:
```suggestion
        "cakephp/orm": "^4.0"
```

File: composer.json [hunk 3]
```diff
@@ -31,11 +31,14 @@
...
     },
     "require-dev": {
-        "cakephp/cakephp": "^3.5",
-        "phpunit/phpunit": "^5.7.14|^6.0",
+        "cakephp/cakephp": "^4.0",
+        "cakephp/cakephp-codesniffer": "^4.0",
+        "phpstan/phpstan": "^0.12",
+        "phpunit/phpunit": "~8.5.0",
+        "psalm/phar": "^3.7",
```
Comment (ADmad) [inline composer.json]:
```suggestion
        "psalm/phar": "^3.9",
```

File: phpstan.neon [hunk 4]
```diff
@@ -0,0 +1,9 @@
+includes:
+    - phpstan-baseline.neon
+
+parameters:
+    level: 6
+    checkGenericClassInNonGenericObjectType: false
+    checkMissingIterableValueType: false
+    excludes_analyse:
+        - */src/TestSuite/*
```
Comment (ADmad) [inline phpstan.neon]: No such folder

File: phpunit.xml.dist [hunk 5]
```diff
@@ -25,6 +26,7 @@
     <filter>
         <whitelist>
             <directory suffix=".php">./src/</directory>
+            <directory suffix=".ctp">./src/</directory>
```
Comment (ADmad) [inline phpunit.xml.dist]: Not applicable in Cake 4

File: src/Model/Behavior/ObfuscateBehavior.php [hunk 6]
```diff
@@ -67,22 +69,25 @@ public function verifyConfig()
...
      * Callback to obfuscate the record(s)' primary key returned after a save operation.
      *
-     * @param \Cake\ORM\Behavior\Event $event Event.
-     * @param \Cake\ORM\Behavior\EntityInterface $entity Entity.
+     * @param \Cake\Event\Event $event Event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity.
      * @param \ArrayObject $options Options.
      * @return void
      */
     public function afterSave(Event $event, EntityInterface $entity, ArrayObject $options)
```
Comment (ADmad) [inline src/Model/Behavior/ObfuscateBehavior.php]: Should use `EventInterface` for all typehints of `Event`.

**Review by ADmad — CHANGES_REQUESTED** [review]
The `CONTRIBUTING.md` has stuff which are not relevant to this repo. There's no `gh-pages` branch, there's no docker file etc. Best to just remove it.

File: .sticker.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+linters:
+  phpcs:
+    standard: CakePHP
```
Comment (ADmad) [inline .sticker.yml]:
```suggestion
    standard: CakePHP4
```

File: .sticker.yml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    extensions: 'php,ctp'
```
Comment (ADmad) [inline .sticker.yml]:
```suggestion
    extensions: 'php'
```

**Pull request merged**